source/github-events/on-new-comments.ts

```diff
@@ -57,10 +44,25 @@ function addListeners(): void {
 	delegates.add(delegate(document, '.js-ajax-pagination', 'submit', paginationSubmitHandler));
 
 	// Collapsed comments are loaded later using an include-fragment element
-	delegates.add(delegate(document, 'details.js-comment-container include-fragment', 'loadstart', getFragmentLoadHandler(run), true));
+	delegates.add(delegate(document, 'details.js-comment-container include-fragment', 'loadstart', getFragmentLoadHandler(run, signal), true));
+}
+
+function removeListeners(): void {
+	for (const subscription of delegates) {
+		subscription.destroy();
+	}
+
+	delegates.clear();
+	handlers.clear();
+	observer.disconnect();
 }
 
-export default function onNewComments(callback: VoidFunction): void {
-	addListeners();
+export default function onNewComments(callback: VoidFunction, signal: AbortSignal): void {
+	if (signal.aborted) {
+		return;
+	}
+
+	addListeners(signal);
 	handlers.add(callback);
+	signal.addEventListener('abort', removeListeners, {once: true});
 }
```

Comment:
This file is awkward. Can it just return a local `delegates` array instead of calling it in its own `removeListeners`

The array would probably also have to include a custom `() => handlers.clear()`